### DISCUSSION TITLE
Double Hotkey.

### DISCUSSION THREAD

**Discussion by ghost** [hyprwm/Hyprland#3958] [Q&A]
Is there a way to set double hotkey, i mean something like Control C + C? so when i press the C by a second time it actually do what i set it to do;

**Comment by imnotpoz** [comment]
https://wiki.hyprland.org/Configuring/Binds/#submaps

in your case something like this:
```
bind=CONTROL,C,submap,my_submap

submap=my_submap

bind=,C,my_action

bind=,escape,submap,reset 

submap=reset
```

**Reply by ghost** [reply]
It did work, but there is a problem, the whole goal of it is to copy the text to the clipboard when i press control + c and then when i press it a second time, it will open an application which would utilize the item copied to the clipboard.

**Reply by vaxerski** [reply]
bindt for the first bind then

**Reply by ghost** [reply]
It still doesn't copy it...

bindt=CONTROL,C,submap,crow

submap=crow

bind=CONTROL,C,exec,dbus-send --type=method_call --dest=io.crow_translate.CrowTranslate /io/crow_translate/CrowTranslate/MainWindow io.crow_translate.CrowTranslate.MainWindow.translateSelection

bind=CONTROL,C,submap,reset 
bind=,ESCAPE,submap,reset 

submap=reset

**Reply by ghost** [reply]
It would be cool if you added another bind(Letter) that includes the ability to handle multiple key presses. It would work as follows:

Syntax:

Bind2 = MODIFIER, KEY, MODIFIER2, KEY2, EXEC, WHATEVER, TIME (1s)

If I press the MODIFIER and KEY combination, the system will wait for the MODIFIER2 and KEY2 within the next second. If these keys are pressed within the specified time frame (especially 1 second), then the command will be executed. If not, it will be ignored.

During execution, it should not act as a global key. For example, if the MODIFIER and KEY are CONTROL and C, it would copy to the clipboard and still await the next key.

**Reply by dapayne12** [reply]
Very late response. But I found something that works in Lua. I have to hit SUPER + C twice to close a window.

```
-- Close window submap: SUPER+C once to arm, SUPER+C again to close; Escape cancels
local closeArmed = false
hl.bind(mainMod .. " + C", function()
    if closeArmed then
        closeArmed = false
        hl.dispatch(hl.dsp.window.close())
        hl.dispatch(hl.dsp.submap("reset"))
    else
        closeArmed = true
        hl.dispatch(hl.dsp.submap("close_confirm"))
    end
end, { submap_universal = true })
hl.define_submap("close_confirm", function()
    hl.bind("escape", function()
        closeArmed = false
        hl.dispatch(hl.dsp.submap("reset"))
    end)
    hl.bind("catchall", hl.dsp.submap("close_confirm"))
end)
```
